utils/src/main/scala/za/co/absa/enceladus/utils/schema/SchemaUtils.scala

```diff
@@ -253,8 +269,28 @@ object SchemaUtils {
     * @param fieldName Name of the field to be appended to the path
     * @return The path with the new field appended or the field itself if path is empty
     */
-  private[enceladus] def appendPath(path: String, fieldName: String) = {
-    if (path.isEmpty) fieldName else s"$path.$fieldName"
+  def appendPath(path: String, fieldName: String): String = {
```

Comment:
Since this is a public method, it might also validate input values. Make sure the input field name does not contain whitespaces or dots, for instance.
There are rules how one can specify a field name with whitespaces. I think '\`' is used (e.g. MY_PATH.\`MY FIELD\`) in such cases. But the conformance engine does not support such fields. I'm pretty sure all nested arrays mapping methods won't work properly with field names contain whitespaces. So maybe just throw an exception if an input column contains one.

Reply:
Since we are trying to solve for Arrays of Arrays. How would this handle if I wanted to send an array index to it? If I remember correctly using `arryName[0]` is the first element of arryName. Do we support this?

Reply:
> 
> 
> Since we are trying to solve for Arrays of Arrays. How would this handle if I wanted to send an array index to it? If I remember correctly using `arryName[0]` is the first element of arryName. Do we support this?

This working on schema, not the actual fields. I believe there is no indexing there. Ans also we are after element types not element values, and those are same withing one array type.